### PR TITLE
Optimize performance of phase field tutorial

### PR DESCRIPTION
**Pull request by ghislainb**
- wrap the algorithm in a `main` function to avoid performance penalty due to the global scope
- use composition operator to define the linear form (which is assembled at each steps)

### CODE REVIEW

File: src/tutorial/phase_field_supercooled.jl [hunk 1]
```diff
@@ -51,106 +51,115 @@ const ny = 20
...
-        write_vtk(
-            joinpath(out_dir, "result_phaseField_imex_1space"),
-            itime,
-            t,
-            mesh,
-            dict_vars;
-            append = true,
-        )
+
+function main()
```
Comment (bmxam) [inline src/tutorial/phase_field_supercooled.jl:57]: could you hide it from literate with `#src` or `#hide`?

File: src/tutorial/phase_field_supercooled.jl [hunk 2]
```diff
@@ -51,106 +51,115 @@ const ny = 20
...
+                joinpath(out_dir, "result_phaseField_imex_1space"),
+                itime,
+                t,
+                mesh,
+                dict_vars;
+                append = true,
+            )
+        end
     end
 end
```
Comment (bmxam) [inline src/tutorial/phase_field_supercooled.jl:159]: could you hide it from literate with `#src` or `#hide`?

File: src/tutorial/phase_field_supercooled.jl [hunk 3]
```diff
@@ -51,106 +51,115 @@ const ny = 20
...
 
+main()
```
Comment (bmxam) [inline src/tutorial/phase_field_supercooled.jl:161]: could you hide it from literate with `#src` or `#hide`?

**Review by lokmanb — APPROVED** [review]
Nice enhancement of the tutorial 🚀

**Comment by bmxam** [comment]
@ghislainb do you want to add some changes or do we merge? I remember that you wanted to put the animation at the top etc.

**Comment by ghislainb** [comment]
> @ghislainb do you want to add some changes or do we merge? I remember that you wanted to put the animation at the top etc.

I think we can merge "as is" (the main purpose of the PR is reached)

**Pull request merged**